Function2 some abstraction leak in teaching curve due to compiler message

Hello, I used rustlings with student this week and I found (and they found it too) that the compiler error in function2 is a little bit complicated as of rustc version 1.46.0 (04488afe3 2020-08-24).

The help and error message give a looooot of information related to rust pre-2018 edition `deprecated annonymous param` and `self` it's great for a seasoned Rustacean but for a beginner like a bachelor 2 student in computer science maybe it's too much.

So I open this issue as a place to possibly think about writing an other erroneous code who do not trigger all this advises but only advises about giving correctly a parameter to `call_me` function.

## Actual code and error + suggestion

```rust
! Compiling of exercises/functions/functions2.rs failed! Please try again. Here's the output:
error: expected one of `:`, `@`, or `|`, found `)`
  --> exercises/functions/functions2.rs:10:15
   |
10 | fn call_me(num) {
   |               ^ expected one of `:`, `@`, or `|`
   |
   = note: anonymous parameters are removed in the 2018 edition (see RFC 1685)
help: if this is a `self` type, give it a parameter name
   |
10 | fn call_me(self: num) {
   |            ^^^^^^^^^
help: if this is a parameter name, give it a type
   |
10 | fn call_me(num: TypeName) {
   |            ^^^^^^^^^^^^^
help: if this is a type, explicitly ignore the parameter name
   |
10 | fn call_me(_: num) {
   |            ^^^^^^

error: aborting due to previous error
```

## Suggestion 1 : change signature from num to num:

```diff
-fn call_me(num) {
+fn call_me(num: ) {
```

Triggers:

```rust
! Compiling of exercises/functions/functions2.rs failed! Please try again. Here's the output:
error: expected type, found `)`
  --> exercises/functions/functions2.rs:10:16
   |
10 | fn call_me(num:) {
   |                ^ expected type

error[E0425]: cannot find value `num` in this scope
  --> exercises/functions/functions2.rs:11:17
   |
11 |     for i in 0..num {
   |                 ^^^ not found in this scope

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0425`.
```

## Suggestion 2 changing `num` to `???` gives:

```diff
-fn call_me(num) {
+fn call_me(???) {
```

```rust
! Compiling of exercises/functions/functions2.rs failed! Please try again. Here's the output:
error: expected parameter name, found `?`
  --> exercises/functions/functions2.rs:10:12
   |
10 | fn call_me(???) {
   |            ^ expected parameter name

error[E0425]: cannot find value `num` in this scope
  --> exercises/functions/functions2.rs:11:17
   |
11 |     for i in 0..num {
   |                 ^^^ not found in this scope

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0425`.
```

Thanks for reading! After the discussion settled to a common point I can do a PR. :)

--- a/exercises/functions/functions2.rs
+++ b/exercises/functions/functions2.rs
@@ -7,7 +7,7 @@ fn main() {
     call_me(3);
 }
 
-fn call_me(num) {
+fn call_me(num:) {
     for i in 0..num {
         println!("Ring! Call number {}", i + 1);
     }

